### PR TITLE
Set default inventory path to /etc/xensource-inventory

### PR DESCRIPTION
**Pull request by lippirk**
I've run into this a couple of times now

### CODE REVIEW

File: lib/inventory.ml [hunk 1]
```diff
@@ -39,6 +39,17 @@ let loaded_inventory = ref false
 let inventory = Hashtbl.create 10
 let inventory_m = Mutex.create ()
 
+let write_to_inventory x =
```
Comment (edwintorok) [inline lib/inventory.ml]: `path` instead of `x` would be more obvious.

File: lib/inventory.ml [hunk 2]
```diff
@@ -39,6 +39,17 @@ let loaded_inventory = ref false
...
+  (** at compile time, inventory_filename is assigned to a path whose parent folder does not exist on a host,
+      so not assigning this variable can give unhelpful error messages, e.g:
+      Unix.Unix_error(Unix.ENOENT, "open", "/usr/lib64/opamroot/ocaml-system/6be9d065-0ada-4ed8-a4af-cc388ea64913")
+    * we advise callers to check that they initialised inventory_filename *)
+  try
+    Unixext.write_string_to_file !inventory_filename x
+  with Unix.(Unix_error (ENOENT, _, _)) as e ->
+   Printf.eprintf "did you initialise 'inventory_filename'?";
```
Comment (edwintorok) [inline lib/inventory.ml]: Do you have access to the debug logging methods here? I'm not sure whether this would go to the right place, or to /dev/null.

**Comment by edwintorok** [comment]
There is a nice ./configure machinery here that allow setting the correct default path, which we nicely set in opam, ... but that causes problems in our actual build system (we use opam in xs-opam), so all that can probably be deleted.

Should the default instead be None, and then we can raise an appropriate error: inventory path not set?

**Comment by lippirk** [comment]
will fix travis also

**Comment by lippirk** [comment]
xapi unit tests are passing with this change

**Pull request merged**